beacon_node/client/src/config.rs

```diff
@@ -97,6 +101,53 @@ impl Config {
             .map(|data_dir| data_dir.join(&self.db_name))
     }
 
+    /// returns whether chain_db was recently purged
+    pub fn chain_db_was_purged(&self) -> bool {
+        return self.get_data_dir().map_or(false,
+            |data_dir| data_dir.join(CHAIN_DB_PURGED_TRAP_FILE).exists());
+    }
+
+    /// purges the chain_db and creates trap file
+    pub fn purge_chain_db(&self) -> Result<(), String> {
+        let data_dir = self.get_data_dir()
+            .ok_or("Failed to get data_dir".to_string())?;
+        let trap_file = data_dir.join(CHAIN_DB_PURGED_TRAP_FILE);
+        fs::File::create(trap_file).map_err( |err| {
```

Comment:
Thinking about the case where the purge fails midway but the trap file has been created: I think it's ok, because the user will probably get errors when the partially deleted database tries to reinitialise. The alternative is creating the trap file afterwards, in which case the user will probably get errors to do with resuming from a deleted DB. In both cases, they'll probably try a re-purge.